homeassistant/components/modbus/climate.py

```diff
@@ -49,7 +48,7 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Read configuration and create Modbus climate."""
-    if discovery_info is None:
+    if discovery_info is None:  # pragma: no cover
```

Comment:
This is an example for `notify` domain but should also work for `climate` domain:
https://github.com/home-assistant/core/blob/418808d873912d84d102909f0ff4e164fac45e27/tests/components/webostv/test_notify.py#L149-L160

Reply:
I was puzzled why it doesn't work, so the reason trying to setup modbus climate as platform doesn't work is because of another bug which prevent setting up the component

When trying to setup modbus climate as platfrom the following error happen and prevent setting up the component:
```
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/setup.py", line 235, in _async_setup_component
    result = await task
  File "/workspaces/core/homeassistant/components/modbus/__init__.py", line 341, in async_setup
    return await async_modbus_setup(
  File "/workspaces/core/homeassistant/components/modbus/modbus.py", line 139, in async_modbus_setup
    for conf_hub in config[DOMAIN]:
KeyError: 'modbus'
```

By adding validation that `modbus` is in config:
before https://github.com/home-assistant/core/blob/aaeaed4117d12ee2e780dfceed53cb52875eec1c/homeassistant/components/modbus/__init__.py#L339
add 
```python
    if DOMAIN not in config:
        return True
```

So that the component now can be set without exception if the user try to set it up like this in YAML:
```yaml
climate:
  - platform: modbus
```
Now the following code will work and get to 100% coverage:
```python
from homeassistant.components.modbus.const import MODBUS_DOMAIN
from homeassistant.setup import async_setup_component


async def test_no_discovery_info(hass, caplog):
    """Test setup without discovery info."""
    assert CLIMATE_DOMAIN not in hass.config.components
    assert await async_setup_component(
        hass,
        CLIMATE_DOMAIN,
        {CLIMATE_DOMAIN: {"platform": MODBUS_DOMAIN}},
    )
    await hass.async_block_till_done()
    assert CLIMATE_DOMAIN in hass.config.components
```

Reply:
But your suggested change does not disturb anything, so I have just add it.

Reply:
That is correct, but a user may still do:
```yaml
climate:
  - platform: modbus
  ```
In this case you have an exception in the code, you don't want to have uncaught exceptions.

adding
```python
  if DOMAIN not in config:
        return True
```
Will prevent this exception, you can add a warning if you want before return true, but it is not needed.

Reply:
Added. Thanks for your suggestion.

Reply:
and you are quite right about the exceptions, so I will add a followup PR that add the new test for the other platforms.